graphql-server/src/schemas/custom.ts

```diff
@@ -758,3 +801,150 @@ async function publishNatsMessage(topic: string, message: string) {
     );
   }
 }
+
+function buildPatientsQueryBody({
+  searchVals,
+  filter,
+}: {
+  searchVals: QueryDashboardPatientsArgs["searchVals"];
+  filter: QueryDashboardPatientsArgs["filter"];
+}) {
+  const fieldsToSearch = [
+    "smilePatientId",
+    "cmoPatientId",
+    "dmpPatientId",
+    "cmoSampleIds",
+    "consentPartA",
+    "consentPartC",
+  ];
+
+  const searchFilters = searchVals?.length
+    ? "WHERE " +
+      fieldsToSearch
+        .map((field) => `${field} =~ '(?i).*(${searchVals.join("|")}).*'`)
+        .join(" OR ")
+    : "";
+
+  const patientsQueryBody = `
+    MATCH (p:Patient)
+
+    // Get patient aliases
+    OPTIONAL MATCH (p)<-[:IS_ALIAS]-(cmoPa:PatientAlias {namespace: 'cmoId'})
+    OPTIONAL MATCH (p)<-[:IS_ALIAS]-(dmpPa:PatientAlias {namespace: 'dmpId'})
+
+    // Get the latest SampleMetadata of each Sample
+    OPTIONAL MATCH (p)-[:HAS_SAMPLE]->(s:Sample)-[:HAS_METADATA]->(sm:SampleMetadata)
+    WITH
+      p,
+      cmoPa,
+      dmpPa,
+      s,
+      collect(sm) AS allSampleMetadata,
+      max(sm.importDate) AS latestImportDate
+    WITH
+      p,
+      cmoPa,
+      dmpPa,
+      s,
+      [sm IN allSampleMetadata WHERE sm.importDate = latestImportDate][0] AS latestSm
```

Comment:
Looks like we are collecting info on what the latest sample metadata is so maybe it _is_ possible to accomplish the "sort by date" after all? As discussed previously though this is not something that's hard required. It would just be a nice bonus. Worth a shot

Reply:
Yep, this is possible. I'll Slack you the results.